pkg/controller/acicontainersoperator.go

```diff
@@ -206,85 +284,85 @@ func (c *Controller) Run(stopCh <-chan struct{}) {
 	}
 
 	// Run informer to start watching and listening
-	go c.Informer.Run(stopCh)
+	go c.Informer_Operator.Run(stopCh)
+	go c.Informer_Deployment.Run(stopCh)
+	go c.Informer_Daemonset.Run(stopCh)
 
 	// Sync the current resources
-	if !cache.WaitForCacheSync(stopCh, c.HasSynced) {
+	if !cache.WaitForCacheSync(stopCh, c.Informer_Operator.HasSynced,
+		c.Informer_Deployment.HasSynced, c.Informer_Daemonset.HasSynced) {
 		utilruntime.HandleError(fmt.Errorf("Controller.Sync: Error syncing the cache"))
 		return
 	}
 	c.Logger.Info("Controller.Sync: Cache sync complete")
 
-	// Run worker in 1 sec interval
-	wait.Until(c.runWorker, time.Second, stopCh)
-}
-
-
-func (c *Controller) HasSynced() bool {
-	return c.Informer.HasSynced()
-}
+	// Run queue for each Informer
+	go c.processQueue(c.Operator_Queue, c.Informer_Operator.GetIndexer(),
+		func(obj interface{}) bool {
+			return c.handleOperatorCreate(obj)
+		},
+		func(obj interface{}) bool {
+			return c.handleOperatorDelete(obj)
+		},
+		stopCh)
 
-// runWorker executes new objects added to queue in loop
-func (c *Controller) runWorker() {
-	log.Info("Controller.runWorker: starting the controller")
+	go c.processQueue(c.Deployment_Queue, c.Informer_Deployment.GetIndexer(),
+		func(obj interface{}) bool {
+			return c.handleDeploymentCreate(obj)
+		}, func(obj interface{}) bool {
+			return c.handleDeploymentDelete(obj)
+		},
+		stopCh)
 
-	for c.processNextItem() {
-		log.Info("Controller.runWorker: processing the next item")
-	}
+	go c.processQueue(c.Daemonset_Queue, c.Informer_Daemonset.GetIndexer(),
+		func(obj interface{}) bool {
+			return c.handleDaemonsetCreate(obj)
+		}, func(obj interface{}) bool {
+			return c.handleDaemonsetDelete(obj)
+		},
+		stopCh)
 
-	log.Info("Controller.runWorker: completed")
 }
 
-func (c *Controller) processNextItem() bool {
-	log.Info("Controller.processNextItem: starting....")
-
-	key, quit := c.Queue.Get()
-
-	if quit {
-		return false
-	}
-
-	defer c.Queue.Done(key)
-
-	keyRaw := key.(string)
-
+func (c *Controller) processQueue(queue workqueue.RateLimitingInterface,
+	store cache.Store, createhandler func(interface{}) bool,
+	deletehandler func(interface{}) bool,
+	stopCh <-chan struct{}) {
+	go wait.Until(func() {
+		log.Info("Starting the handlers....")
+		for {
+			key, quit := queue.Get()
+			if quit {
+				break
+			}
+			var requeue bool
+			switch key := key.(type) {
+			case chan struct{}:
+				close(key)
+			case string:
+				obj, exists, err := store.GetByKey(key)
+				if err == nil && exists {
+					log.Info("Controller.processNextItem: object Creation detected:", key)
+					requeue = createhandler(obj)
+				}
+
+				if !exists {
+					log.Info("Controller.processNextItem: object deleted detected:", key)
+					deletehandler(key)
```

Comment:
do you want to handle later your delete handler ? as you are not doing anything in the dunction

Reply:
Currently, yes it is only logging information about whether the object is deleted or not. I am using it as an endpoint if in future we want anything in deletehandler